syntax = "proto3";
package pt.ulisboa.tecnico.nameserver.contract;

message RegisterRequest {
  string serviceName = 1;
  string qualifier = 2;
  string target = 3;
}
message RegisterResponse {
}

message LookupRequest {
  string serviceName = 1;
  string qualifier = 2;
}

message LookupResponse {
  repeated string serverList = 1;
}

message DeleteRequest {
  string serviceName = 1;
  string target = 2;
  
}

message DeleteResponse {
}

service NameServer {
  rpc register(RegisterRequest) returns (RegisterResponse);
  rpc lookup(LookupRequest) returns (LookupResponse);
  rpc delete(DeleteRequest) returns (DeleteResponse);
}